### PR TITLE
Support text search

### PR DESCRIPTION
**Pull request by bright-starry-sky**
Depends on:
 https://github.com/vesoft-inc/nebula-common/pull/260

### CODE REVIEW

File: src/executor/admin/SpaceExecutor.cpp [hunk 1]
```diff
@@ -10,24 +10,62 @@
...
+                        return Status::Error("external index create failed : %s",
+                                             edgeTSIndex.c_str());
+                    }
+
+                    auto tagTSIndex = nebula::plugin::IndexTraits::indexName(spaceName, false);
+                    if (createTSIndex(ret.value(), tagTSIndex)) {
+                        return Status::Error("external index create failed : %s",
+                                             tagTSIndex.c_str());
+                    }
+                }
```
Comment (critical27) [inline src/executor/admin/SpaceExecutor.cpp:46]: I am not pretty sure we should create index here. Why we don't `CREATE FULLTEXT TAG | EDGE INDEX index_name ON space_name [fulltext_clause]` like in design doc?

File: src/parser/parser.yy [hunk 2]
```diff
@@ -1048,8 +1073,245 @@ match_return
...
+    }
+    ;
+
+sign_out_text_search_service_sentence
+    : KW_SIGN KW_OUT KW_TEXT KW_SERVICE {
+        $$ = new SignOutTextServiceSentence();
+    }
+    ;
+
+text_search_argument
```
Comment (critical27) [inline src/parser/parser.yy:1120]: We could add some doc later of how to use full text index

File: src/validator/IndexScanValidator.cpp [hunk 3]
```diff
@@ -83,15 +97,148 @@ Status IndexScanValidator::prepareYield() {
...
+                if (expr->arg()->fuzziness() == -1) {
+                    fuzz = "AUTO";
+                } else {
+                    fuzz = expr->arg()->fuzziness();
+                }
+                suc = nebula::plugin::ESGraphAdapter::kAdapter->fuzzy(ftc,
+                                                                      doc,
+                                                                      limit,
+                                                                      fuzz,
+                                                                      *expr->arg()->op(),
```
Comment (critical27) [inline src/validator/IndexScanValidator.cpp:203]: What is op, and/or?

File: src/executor/admin/SpaceExecutor.cpp (hunk 1 above)
Comment (bright-starry-sky) [reply src/executor/admin/SpaceExecutor.cpp:46]: > I am not pretty sure we should create index here. Why we don't `CREATE FULLTEXT TAG | EDGE INDEX index_name ON space_name [fulltext_clause]` like in design doc?

Good question. the original design is similar to what you said.after an offline discussion with Sherman. I modified it to create external indexes and paces together. For two reasons: 1, external indexes are associated with space, not with tag or edge; 2, external index creation does not require meta; the graph layer itself is ok.

File: src/validator/IndexScanValidator.cpp (hunk 3 above)
Comment (bright-starry-sky) [reply src/validator/IndexScanValidator.cpp:203]: > What is op, and/or?

Yes, same with match api of ElasticSearch.

**Comment by bright-starry-sky** [comment]
Done. ready to review.

File: src/validator/IndexScanValidator.cpp [hunk 4]
```diff
@@ -93,26 +95,173 @@ Status IndexScanValidator::prepareFilter() {
...
+        return relExpr;
+    } else {
+        return genOrFilterFromList(tsExpr, vRet.value());
+    }
+    return Status::OK();
+}
+
+std::unique_ptr<Expression> IndexScanValidator::genOrFilterFromList(
+    TextSearchExpression* expr, const std::vector<std::string>& values) {
+    auto filter = std::make_unique<LogicalExpression>(
```
Comment (critical27) [inline src/validator/IndexScanValidator.cpp:151]: It seems `LogicalExpression` has support multiple operands, could we use it?

Comment (critical27) [reply src/validator/IndexScanValidator.cpp:151]: BTW, have test the performance between pass as filter or pass by  multi`IndexQueryContext`

**Pull request closed without merging**